hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AtttributeResolver.kt

```diff
@@ -2,14 +2,21 @@ package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesea
 
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.data.elasticsearch.annotations.Field
 import uk.gov.justice.digital.hmpps.prisonersearch.common.model.Prisoner
 import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.TypeMatcher
 import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.genericType
 import kotlin.reflect.KClass
 import kotlin.reflect.KProperty1
 import kotlin.reflect.full.memberProperties
+import kotlin.reflect.jvm.javaField
 
-typealias Attributes = Map<String, KClass<*>>
+typealias Attributes = Map<String, Attribute>
+
+class Attribute(
+  val type: KClass<*>,
+  val openSearchName: String,
+)
```

Comment:
This is the main change - previously we were trying to work out which fields should be appended with ".keyword" on an ad-hoc basis. We now do that up front in this class by checking for the annotation `@Field(type = FieldType.Keyword)`. This keeps the solution generic and fixes bugs for fields that couldn't be searched.